v6_carrier_py/master.py

```diff
@@ -105,12 +112,53 @@ def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args,
     If no keys are specified the datasets are joined on all columns with the same name.
     TODO: What if different datasets use different keys to mean the same thing? How do we specify this?
     """
-    results = _dispatch_tasks(client, data, *args, method='get_data', **kwargs)
+    combined_df = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+    return combined_df.corr()
+
+
+def train_model(client: ClientContainerProtocol, data, pipeline: Pipeline, features: List[str], target: str, keys=None,
+                *args, **kwargs):
+    """
+    Retrieve data from nodes and train data analysis pipeline on it. Returns the performance of the resulting model.
+    TODO: How and where do we save our model?
+    """
+    try:
+        info(f'Training pipeline with the following steps: {pipeline.named_steps}')
+        results = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+        X = results[features].values
+        y = results[target].values
+
+        # Split data
+        X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+        pipeline.fit(X_train, y_train)
+        predictions = pipeline.predict(X_test)
+
+        # TODO: Make metrics configurable
+        score = metrics.mean_absolute_error(y_test, predictions)
+
+        return score
+    except Exception as e:
+        # Log stacktrace
+        traceback.print_exc()
+
+
+def _combine_all_node_data(client, data, keys, *args, **kwargs) -> pd.DataFrame:
```

Comment:
Can you rename `keys` to something like: `record_identifiers` or `identifying_columns`. Something that says what it is.

Reply:
If you by the way want to enforce pseudonymization this will always be the encrypted identifier column